lib/cart.js

```diff
@@ -0,0 +1,67 @@
+const decodeBase64Id = (encodedId) => {
+  // Some IDs are like Z2lkOi8A3Mzg5MTM4Nw==::706db06f-4ce4-4f
+  // We extract the base64 part from these.
+  // After decoding, we can get strings like gid://shopify/variant/3423489374893
+
+  if (encodedId && typeof encodedId === 'string') {
+    const base64id = encodedId.split('::').shift()
+    return Buffer.from(base64id, 'base64').toString('binary').split('/').pop()
+  }
+  return encodedId
+}
+
+module.exports = {
+  saveCart: async (cart, products, ctx) => {
+    if (!cart.lineItems) return
+
+    const cartData = cart.lineItems.reduce(function (sum, item) {
+      sum.itemCount += Number(item.quantity)
+      sum.value += Number(item.variant.price) * Number(item.quantity)
+      return sum
+    }, { itemCount: 0, value: 0, returnUrl: window.location.hostname })
+
+    if (cart.lineItems.length > 0) {
+      cartData.currency = cart.lineItems[0].variant.priceCurrency
+    }
+
+    const cartItems = []
+    for (const lineItem of cart.lineItems) {
+      const quantity = Number(lineItem.quantity)
+      const price = Number(lineItem.variant.price)
+
+      // sometimes we can get it from the state, but it's a deprecated feature.
+      let product = products &&
+        products.products &&
+        products.products[lineItem.handle] &&
+        products.products[lineItem.handle].product
```

Comment:
We could add `lodash.get` to reduce the unnecessary complexity of the code

(or add a transpiler and use conditional chaining, but we agreed not to do that for now)

Reply:
And use it for all the `.variant.something` occurrences too